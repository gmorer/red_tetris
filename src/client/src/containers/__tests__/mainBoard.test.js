// import lel from './../mainBoard'
import component from './../mainBoard'
import React from 'react'
import renderer from 'react-test-renderer'

const piecesArray = [2, 3, 4, 5, 6];

const { Board, deleteEmptyRow, getNextPiece } = component

test('mainboard test', () => {
	const component = renderer.create(<Board piecesArray={piecesArray} boards={[]} />)
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
})

test('deletEmptyRow test', () => {
	expect(deleteEmptyRow([
		[' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
		[' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
		[' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
		[' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
		[' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
		[' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '],
		[' ', 'x', 'x', 'x', 'x', 'x', ' ', ' ', ' ', ' '],
		['x', 'x', 'x', 'x', 'x', 'x', 'x', 'x', 'x', 'x'],
		['x', 'x', 'x', 'x', 'x', ' ', 'x', 'x', 'x', 'x'],
		['x', 'x', 'x', 'x', 'x', 'x', 'x', 'x', 'x', 'x']
	])).toBe(2);
})

test('getNextPiece test', () => {
	expect(getNextPiece(new Array(7), 4)).toBe(5)
	expect(getNextPiece(new Array(7), 6)).toBe(0)
})