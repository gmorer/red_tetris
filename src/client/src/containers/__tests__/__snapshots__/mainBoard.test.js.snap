// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainboard test 1`] = `
<div
  style={
    Object {
      "backgroundImage": "url('bg2.jpg')",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "100% 100%",
      "display": "flex",
      "height": "100%",
      "justifyContent": "space-around",
      "paddingTop": "2em",
      "position": "relative",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <h1
      style={
        Object {
          "color": "black",
        }
      }
    />
    <div
      style={
        Object {
          "display": "flex",
          "height": "100%",
          "justifyContent": "space-evenly",
        }
      }
    >
      <div
        id="mainBoard"
      >
        <div
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0,  0.5)",
              "height": "100%",
              "position": "absolute",
              "textAlign": "center",
              "width": "100%",
              "zIndex": "99",
            }
          }
        >
          <h3
            style={
              Object {
                "color": "red",
                "fontSize": "3em",
                "marginTop": "50%",
              }
            }
          >
            Game
            <br />
            Over
          </h3>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "height": "100%",
              "marginTop": "-40%",
              "width": "100%",
            }
          }
        />
      </div>
      <div>
        <div
          style={
            Object {
              "backgroundColor": "rgba(159, 35, 58, 0.5)",
              "height": "10em",
              "width": "10em",
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "height": "100%",
                "left": "0%",
                "position": "relative",
                "top": "0%",
                "width": "100%",
              }
            }
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": "25%",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": "25%",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "border": "2px solid #996300",
                    "boxSizing": "border-box",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "borderColor": "#FFA500 red red #ffc600",
                        "borderStyle": "solid",
                        "borderWidth": "10em 0em 0em 10em",
                        "height": "0px",
                        "width": "0px",
                      }
                    }
                  />
                </div>
              </div>
              <div
                style={
                  Object {
                    "border": "2px solid #996300",
                    "boxSizing": "border-box",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "borderColor": "#FFA500 red red #ffc600",
                        "borderStyle": "solid",
                        "borderWidth": "10em 0em 0em 10em",
                        "height": "0px",
                        "width": "0px",
                      }
                    }
                  />
                </div>
              </div>
              <div
                style={
                  Object {
                    "border": "2px solid #996300",
                    "boxSizing": "border-box",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "borderColor": "#FFA500 red red #ffc600",
                        "borderStyle": "solid",
                        "borderWidth": "10em 0em 0em 10em",
                        "height": "0px",
                        "width": "0px",
                      }
                    }
                  />
                </div>
              </div>
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": "25%",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "border": "2px solid #996300",
                    "boxSizing": "border-box",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "borderColor": "#FFA500 red red #ffc600",
                        "borderStyle": "solid",
                        "borderWidth": "10em 0em 0em 10em",
                        "height": "0px",
                        "width": "0px",
                      }
                    }
                  />
                </div>
              </div>
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": "25%",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
              <div
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "25%",
                  }
                }
              />
            </div>
          </div>
          <p>
            Lambda
          </p>
        </div>
        <div
          style={
            Object {
              "backgroundColor": "rgba(159, 35, 58, 0.5)",
              "height": "5em",
              "width": "10em",
            }
          }
        >
          Score : 
          0
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "height": "80%",
        "overflowY": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "alignSelf": "center",
          "display": "flex",
          "flexFlow": "row wrap",
          "justifyContent": "space-around",
          "marginRight": "3em",
          "marginTop": "5em",
        }
      }
    />
  </div>
</div>
`;
